### PR TITLE
#159143845 User should save a game, players and those that left agame

### PR DESCRIPTION
**Pull request by kenware**
#### What does this PR do?
-Saves a game to the database
#### Description of Task to be completed?
-Save game history if there is authenticated user  that played the game
-Saves the game winner,
-Saves the players 

-Saves the players that left a game
#### How should this be manually tested?
-Navigate to the sagging branch in Heroku
-Sign up and Play the game 
-After the game ends, your game history will be saved to the DB
https://et-cfh-staging-pr-24.herokuapp.com/#!/
#### What are the relevant pivotal tracker stories?
[#159143845]

### CODE REVIEW

File: public/js/services/game.js [hunk 1]
```diff
@@ -119,17 +119,17 @@ angular.module('mean.system')
       if (data.table.length === 0) {
         game.table = [];
       } else {
-        var added = _.difference(_.pluck(data.table, 'player'), _.pluck(game.table, 'player'));
-        var removed = _.difference(_.pluck(game.table, 'player'), _.pluck(data.table, 'player'));
-        for (i = 0; i < added.length; i++) {
-          for (var j = 0; j < data.table.length; j++) {
+        const added = _.difference(_.pluck(data.table, 'player'), _.pluck(game.table, 'player'));
+        const removed = _.difference(_.pluck(game.table, 'player'), _.pluck(data.table, 'player'));
```
[the remark this answers is not in the document]
Comment (kenware) [reply public/js/services/game.js:123]: There is no way to define underscore in frontend as it is from CDN link

File: public/js/services/game.js [hunk 2]
```diff
@@ -119,17 +119,17 @@ angular.module('mean.system')
       if (data.table.length === 0) {
         game.table = [];
       } else {
-        var added = _.difference(_.pluck(data.table, 'player'), _.pluck(game.table, 'player'));
-        var removed = _.difference(_.pluck(game.table, 'player'), _.pluck(data.table, 'player'));
-        for (i = 0; i < added.length; i++) {
-          for (var j = 0; j < data.table.length; j++) {
+        const added = _.difference(_.pluck(data.table, 'player'), _.pluck(game.table, 'player'));
```
[the remark this answers is not in the document]
Comment (kenware) [reply public/js/services/game.js:122]: There is no way to define underscore in frontend as it is from CDN link

File: config/socket/game.js [hunk 3]
```diff
@@ -1,25 +1,36 @@
...
-  "The Spleen",
-  "Dingle Dangle"
+/* eslint func-names: 0, prefer-arrow-callback: 0 */
+import async from 'async';
+import _ from 'underscore';
+import mongoose from 'mongoose';
+
+const GameModel = mongoose.model('Game');
+const questions = require(`${__dirname}/../../app/controllers/questions.js`);
+const answers = require(`${__dirname}/../../app/controllers/answers.js`);
```
[the remark this answers is not in the document]
Comment (kenware) [reply config/socket/game.js]: Dynamic require does not work with es6 import

File: config/socket/game.js [hunk 4]
```diff
@@ -1,25 +1,36 @@
...
-  "Swollen Node",
-  "The Spleen",
-  "Dingle Dangle"
+/* eslint func-names: 0, prefer-arrow-callback: 0 */
+import async from 'async';
+import _ from 'underscore';
+import mongoose from 'mongoose';
+
+const GameModel = mongoose.model('Game');
+const questions = require(`${__dirname}/../../app/controllers/questions.js`);
```
[the remark this answers is not in the document]
Comment (kenware) [reply config/socket/game.js]: Dynamic require does not work with es6 import

File: public/js/services/game.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (kenware) [reply public/js/services/game.js:123]: There is no way to define underscore in frontend as it is from CDN link

File: config/socket/game.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (kenware) [reply config/socket/game.js]: There is no way to define underscore in frontend as it is from CDN link

File: config/socket/game.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (kenware) [reply config/socket/game.js]: There is no way to define underscore in frontend as it is from CDN link

**Review by adetola-ralph — APPROVED** [review]

**Pull request merged**